LanguageFeatures/Extension-types/exhaustiveness_variable_A04_t01.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Exhaustiveness of a variable pattern is determined by the static
+/// type of the corresponding variable.
+///
+/// @description Check static type of a variable pattern
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import '../../Utils/static_type_helper.dart';
+
+extension type ET(int i) {}
+extension type TO(int i) implements ET {}
+
+main() {
+  ET x = ET(1);
+  switch (x) {
+    case int y:
+      x.expectStaticType<Exactly<ET>>;
```

Comment:
Yep! This is the kind of test I was looking for above: The type of `x` hasn't been erased to `int`.